src/CohortManager.js

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable array-callback-return */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-unused-vars */
+const Cohort = require('./Cohort')
+const Student = require('./Student')
+
+class CohortManager {
+  constructor() {
+    this.cohorts = []
+    this.students = []
+    this.studentsId = 1
+  }
+
+  addCohort(name) {
+    const regex = /^[A-Za-z0-9 ]*$/
+    if (name === '') {
+      return 'invalid name'
+    }
+    if (regex.test(name)) {
+      const newCohort = new Cohort(name)
+      const selectedName = []
+
+      this.cohorts.push(newCohort)
+
+      this.cohorts.find((element) => {
+        if (element.name === name) {
+          selectedName.push(element)
+        }
+      })
+      return selectedName.length ? 'cohort already exist' : this.cohorts
+    }
+    return 'invalid name'
+  }
+
+  addStudent(name, lastname, email, gitLink, id) {
+    id = this.studentsId++
+    const newStudent = new Student(name, lastname, gitLink, email, id)
+    this.students.push(newStudent)
+    return newStudent
+  }
+
+  searchByName(list, name) {
```

Comment:
I think a better method here would be `searchCohortsByName(name)` - make it do one thing and do it well